share/p2p/peers/manager.go

```diff
@@ -182,13 +210,11 @@ func (m *Manager) Peer(
 	// first, check if a peer is available for the given datahash
 	peerID, ok := p.tryGet()
 	if ok {
-		// some pools could still have blacklisted peers in storage
-		if m.isBlacklistedPeer(peerID) {
-			log.Debugw("removing blacklisted peer from pool",
-				"peer", peerID.String())
-			p.remove(peerID)
+		if m.removeUnreachable(p, peerID) {
```

Comment:
nit: For the third time I read this PR and every time I misinterpret what we removing here.

What if name it as `removeIfUnreachable`, little change but improves readability.

Reply:
fixed